content/markdown/02-working-with-data/02-01-reading-writing.rmd

```diff
@@ -0,0 +1,147 @@
+
+We can read in the file by assinging it directly to an object.
+
+**Base R**
+```{r, eval=FALSE}
+data <- read.csv("/path/to/file/file-name.csv", header = TRUE)
+```
+
+Here the first argument in the `read.csv()` function is the path pointing to the file in your computer. The second argument specifies `header = TRUE` which tells R that the first row of the `.csv` file will be the column names, not part of the data themselves. 
+
+**Tidyverse**
+
+```{r, eval=FALSE}
+data <- readr::read_csv("/path/to/file/file-name.csv")
+```
+
+### Excel Sheets `.xls(x)`
+
+The best way to read in an excel sheet into R with with the `readxl` package. Almost identical to above: 
+
+```{r, eval=FALSE}
+data <- readxl::read_xls("/path/to/file/file-name.csv", sheet = "SHEETNAME")
+```
+
+Here, the path argument is identical above, but the `sheet =` argument is an optional argument that you can pass the name of a particular sheet in the file if you only want to read in one sheet out of many. 
+
+### Plain Text `.txt`
+
+Reading a text file is similar to the approaches to a csv file: 
+
+**Base R**
+
+```{r, eval=FALSE}
+data <- read.delim("/path/to/file/file-name.csv")
+```
+
+This above option assumes that the file it tab-delimited. 
+
+**Tidyverse**
+
+```{r, eval=FALSE}
+data <- readr::read_tsv("/path/to/file/file-name.csv")
+```
+
+## Writing Files
+
+Writing files is nearly identical to reading files. 
+
+### Comma-Separated Value `.csv` Files 
+
+**core R**
+
+```{r, eval=FALSE}
+write.csv(data_object, "/path/to/new/file/file-name.csv")
+```
+
+**Tidyverse**
+```{r, eval=FALSE}
+readr::write_csv(data_object, "/path/to/new/file/file-name.csv")
+```
+
+### Excel Sheets `.xls(x)`
+
+Writing a file to an excel sheet is slightly unusual and wouldn't be recommended (`.csv` files are always better), but if you need to, there exists a number of packages to assist, the easiest of which being `writexl`.
+
+
+```{r, eval=FALSE}
+library(writexl)
+writexl::write_xlsx(data_object, "/path/to/new/file/file-name.xlsx")
+```
+
+### Plain Text `.txt`
+
+**core R** 
+
+```{r, eval=FALSE}
+write.table(data_object, "/path/to/new/file/file-name.txt")
+```
+
+**Tidyverse**
+
+```{r, eval=FALSE}
+readr::write_delim(data_object, "/path/to/new/file/file-name.txt")
+```
+
+## Writing Figures
+
+It is common to create a figure in R and want to save it to the local machine. If you are plotting in `ggplot2`, then there are specific functions for those objects, but if you are plotting in base R, you will use built-in functions to save as whatever file type you want. 
+
+Using the example built-in dataset `airquality`, we can make an example plot to save. 
+
+### Saving an image created in base R 
+
+```{r}
+plot(x = airquality$Day, y = airquality$Temp, col = "red")
+```
+If this is the plot we wish to save, we can save it as any common figure file type (e.g. `.png`, `.jpg`, `.tif`, `.pdf` etc.)
+
+```{r, eval=FALSE}
+# first use the command for the file type 
```

Comment:
you might want to say something like- the first line opens the connection to a file (makes an empty spot), the plot function makes the plot, and the the last line writes the file and closes the connection